validator/sawtooth_validator/consensus/handlers.py

```
@@ -132,7 +132,14 @@ def __init__(self, proxy):
         self._proxy = proxy
 
     def handle_request(self, request, response):
-        self._proxy.send_to(request.peer_id, request.message)
+        try:
+            self._proxy.send_to(
+                request.peer_id,
+                request.message.SerializeToString())
```

The proxy should raise some exception if the peer_id is not found and the handler should catch this and return an error to the engine.

---

Agreed, but this will require modifying the interfaces to gossip and/or interconnect.